otherlibs/unix/select_unix.c

```diff
@@ -79,12 +82,14 @@ CAMLprim value caml_unix_select(value readfds, value writefds, value exceptfds,
   retcode += fdlist_to_fdset(exceptfds, &except, &maxfd);
   /* PR#5563: if a bad fd was encountered, report EINVAL error */
   if (retcode != 0) caml_unix_error(EINVAL, "select", Nothing);
-  tm = Double_val(timeout);
-  if (tm < 0.0)
+  tm_sec = Double_val(timeout_sec);
+  if (tm_sec < 0.0)
     tvp = (struct timeval *) NULL;
   else {
-    tv.tv_sec = (int) tm;
-    tv.tv_usec = (int) (1e6 * (tm - tv.tv_sec));
+    double int_sec, frac_sec;
+    frac_sec = modf(tm_sec, &int_sec);
+    tv.tv_sec = int_sec;
+    tv.tv_usec = frac_sec * USEC_PER_SEC;
```

Comment:
Just to check my understanding. Before we had a cast for `tv_usec`, and now you are using implicit cast rules from the double-ish `frac_sec` value and the int-ish `tv_usec` field, right?

Reply:
Yes, I'm using implicit cast rules here.